ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -342,86 +378,120 @@ class TreegraphSeries extends OrganizationSeries {
+        id: string,
+        index: number,
+        level: number,
+        list: Record<string, number[]>,
+        parent?: string
+    ): this['tree'] {
+        let point = this.points[index];
+        level = (point && point.level) || level;
+        return super.buildTree.call(this, id, index, level, list, parent);
+    }
+
     /**
      * Run pre-translation by generating the nodeColumns.
      * @private
      */
     public translate(): void {
-        if (!this.processedXData) {
-            this.processData();
-        }
-        this.generatePoints();
-        this.nodeColumns = this.createNodeColumns();
-
-        const series = this,
-            options = this.options,
-            nodeColumns = this.nodeColumns;
+        let series = this,
+            options = series.options,
+            // NOTE: updateRootId modifies series.
+            rootId = TU.updateRootId(series),
+            rootNode,
+            tree: TreegraphNode.Node;
+        // Call prototype function
+        Series.prototype.translate.call(series);
+
+        // @todo Only if series.isDirtyData is true
+        tree = series.tree = series.getTree();
+
+        series.links.forEach((link): void => {
+            link.destroy();
+        });
+        series.links = series.getLinks();
+        rootNode = series.nodeMap[rootId];
 
-        this.nodePadding = this.getNodePadding();
+        if (rootId !== '' && (!rootNode || !rootNode.children.length)) {
+            series.setRootNode('', false);
+            rootId = series.rootNode;
+            rootNode = series.nodeMap[rootId];
+        }
 
-        // Calculate level options used in sankey, organization, and treegraph.
-        series.mapOptionsToLevel = getLevelOptions({
-            // NOTE: if support for allowTraversingTree is added, then from
-            // should be the level of the root node.
-            from: 1,
+        series.mapOptionsToLevel = getLevelOptions<this>({
+            from: rootNode.level + 1,
             levels: options.levels,
-            to: nodeColumns.length - 1, // Height of the tree
+            to: tree.height,
             defaults: {
-                borderColor: options.borderColor,
-                borderRadius: options.borderRadius, // organization series
-                borderWidth: options.borderWidth,
-                color: series.color,
-                colorByPoint: options.colorByPoint,
-                // NOTE: if support for allowTraversingTree is added, then
-                // levelIsConstant should be optional.
-                levelIsConstant: true,
-                linkColor: options.linkColor, // organization series
-                linkLineWidth: options.linkLineWidth, // organization series
-                linkOpacity: options.linkOpacity,
-                states: options.states
+                levelIsConstant: series.options.levelIsConstant,
+                colorByPoint: options.colorByPoint
             }
-        });
+        }) as any;
```

Comment:
`as any` required?

Reply:
copied from `TreemapSeries.ts` so it might need to be resolved higher in the inheritance chain, and I didn't want to resolve all of the issues of the `treemap` at once ;) it can be changed tho if you want.